channel/channel.go

```diff
@@ -0,0 +1,126 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package channel
+
+import (
+	"context"
+	"crypto/x509"
+	"github.com/openziti/foundation/identity/identity"
+	"github.com/openziti/foundation/transport"
+	"io"
+	"time"
+)
+
+// Channel represents an asyncronous, message-passing framework, designed to sit on top of an underlay.
+//
+type Channel interface {
+	Identity
+	SetLogicalName(logicalName string)
+	Binding
+	Sender
+	io.Closer
+	IsClosed() bool
+	Underlay() Underlay
+	StartRx()
+	GetTimeSinceLastRead() time.Duration
+}
+
```

Comment:
Talked with Andrew about some ways to improve this API in a second pass. Will likely move some things into Message directly to simplify the API more and split things that are called by API consumers from things called by API internals.

Would also like to split channel out into its own package, both for modularity and so it's easier to version, since we can then use module versioning without having to bump all of foundation.